tap_postgres/sync_strategies/logical_replication.py

```diff
@@ -217,14 +217,90 @@ def row_to_singer_message(stream, row, version, columns, time_extracted, md_map,
+        yield None
+    else:
+        tap_stream_id = post_db.compute_tap_stream_id(conn_info['dbname'], payload['schema'], payload['table'])
+        if streams_lookup.get(tap_stream_id) is None:
+            yield None
+        else:
+            target_stream = streams_lookup[tap_stream_id]
+            stream_version = get_stream_version(target_stream['tap_stream_id'], state)
+            stream_md_map = metadata.to_map(target_stream['metadata'])
+
+            desired_columns = [col for col in target_stream['schema']['properties'].keys() if sync_common.should_sync_column(stream_md_map, col)]
+
+            if payload['action'] == 'I':
+                col_names = []
+                col_vals = []
+
+                for column in payload['columns']:
+                    if column['name'] in set(desired_columns):
+                        col_names.append(column['name'])
+                        col_vals.append(column['value'])
+
+                col_names = col_names + ['_sdc_deleted_at']
+                col_vals = col_vals + [None]
+                if conn_info.get('debug_lsn'):
+                    col_names = col_names + ['_sdc_lsn']
+                    col_vals = col_vals + [str(lsn)]
+                record_message = row_to_singer_message(target_stream, col_vals, stream_version, col_names, time_extracted, stream_md_map, conn_info)
+
+            elif payload['action'] == 'U':
+                col_names = []
+                col_vals = []
+                for column in payload['columns']:
+                    if column['name'] in set(desired_columns):
+                        col_names.append(column['name'])
+                        col_vals.append(column['value'])
+
+                col_names = col_names + ['_sdc_deleted_at']
+                col_vals = col_vals + [None]
+
+                if conn_info.get('debug_lsn'):
+                    col_vals = col_vals + [str(lsn)]
+                    col_names = col_names + ['_sdc_lsn']
+                record_message = row_to_singer_message(target_stream, col_vals, stream_version, col_names, time_extracted, stream_md_map, conn_info)
+
+            elif payload['action'] == 'D':
+                col_names = []
+                col_vals = []
+
+                for column in payload['identity']:
+                    if column['name'] in set(desired_columns):
+                        col_names.append(column['name'])
+                        col_vals.append(column['value'])
+
+                col_names = col_names + ['_sdc_deleted_at']
+                col_vals = col_vals + [singer.utils.strftime(singer.utils.strptime_to_utc(payload['timestamp']))]
+                if conn_info.get('debug_lsn'):
+                    col_vals = col_vals + [str(lsn)]
+                    col_names = col_names + ['_sdc_lsn']
+                record_message = row_to_singer_message(target_stream, col_vals, stream_version, col_names, time_extracted, stream_md_map, conn_info)
```

Comment:
I'd prefer to keep it here so it's easier to refactor later on.

Reply:
Do we anticipate needing refactoring this later? Why not refactor it now?

Reply:
I'm apprehensive to go too crazy on this for the first round since we'll be learning about how different databases react to it over time. So keeping it explicitly connected to help readability and fix initial bugs is my goal. Rather than try to wrap it up into an assumed pattern that we don't yet understand fully due to no one using it in practice yet.

Reply:
Actually, looking at it, pulling the call down to the yield will keep it in a spot where it'd be less likely to cause a bug if changed. I'll do that